src/test/java/io/castle/client/internal/utils/CastleContextBuilderTest.java

```diff
@@ -247,6 +248,29 @@ public void withManualHeaders() throws CastleSdkConfigurationException, JSONExce
         JSONAssert.assertEquals(contextJson, "{\"active\":true,\"headers\":{\"User-Agent\":\"ua\"}," + SDKVersion.getLibraryString() + "}", false);
     }
 
+    @Test
+    public void withCustomIPHeaders() throws CastleSdkConfigurationException, JSONException {
+        // Given
+        CastleConfiguration configuration = CastleConfigurationBuilder.defaultConfigBuilder()
+                .apiSecret("abcd")
+                .ipHeaders(Arrays.asList("X-Forwarded-For", "CF-Connecting-IP"))
+                .build();
+
+        MockHttpServletRequest standardRequest = getStandardRequestMock();
+        standardRequest.addHeader("X-Forwarded-For", "1.1.1.1");
```

Comment:
Worth noting that `X-Forwarded-For` will normally not look like that `1.1.1.1`, since it's used to keep a list of all IPs involved. It will most commonly look like `client, proxy1, proxy2, etc`.

Reply:
Updated the test to a realistic value